pkgs/http_client_conformance_tests/lib/src/server_queue_helpers.dart

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:async/async.dart';
+
+extension StreamQueueOfNullableObjectExtension on StreamQueue<Object?> {
+  Future<int> get nextAsInt async => ((await next) as num).toInt();
```

Comment:
Could you add something like:
// When run under dart2wasm, JSON numbers are always returned floats.

And please file an issue to run the tests under dart2wasm.

Reply:
Done!